### PR TITLE
feat: allow json serializer to generate a json array

### PR DESCRIPTION
**Pull request by myalamov**
## Summary

On batch serialization mode the JSON former adds a metrics field containing the metric array (JSON object containing a key with an array of objects)
```
{
    "metrics": [
        {
            "fields": {
                "field_1": 30,
                "field_2": 4,
                "field_N": 59,
                "n_images": 660
            },
            "name": "docker",
            "tags": {
                "host": "raynor"
            },
            "timestamp": 1458229140
        },
        {
            "fields": {
                "field_1": 30,
                "field_2": 4,
                "field_N": 59,
                "n_images": 660
            },
            "name": "docker",
            "tags": {
                "host": "raynor"
            },
            "timestamp": 1458229140
        }
    ]
}
```
The change adds a configuration option **output_json_array** which outputs a JSON array:
```
[
  {
      "fields": {
          "field_1": 30,
          "field_2": 4,
          "field_N": 59,
          "n_images": 660
      },
      "name": "docker",
      "tags": {
          "host": "raynor"
      },
      "timestamp": 1458229140
  },
  {
      "fields": {
          "field_1": 30,
          "field_2": 4,
          "field_N": 59,
          "n_images": 660
      },
      "name": "docker",
      "tags": {
          "host": "raynor"
      },
      "timestamp": 1458229140
  }
]
```
That will allow some database servers like Clickhouse to ingesting the data in a batches.
Sample usage:
```
      [[outputs.http]]
        url = "http://clickhouse.server.com:8123/?query=INSERT%20INTO%20_TABLENAME_%20FORMAT%20JSONEachRow"
        username = "sample_username"
        password = "sample_password"
        method = "POST"
        data_format = "json"
        output_json_array = true
        [outputs.http.headers]
          Content-Type = "application/json"
```

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #

### CODE REVIEW

**Comment by myalamov** [comment]
!signed-cla

File: plugins/serializers/json/README.md [hunk 1]
```diff
@@ -99,6 +102,42 @@ reference the documentation for the specific plugin.
 }
 ```
 
+When batch forma is used and the configuration option output_json_array
```
Comment (DStrand1) [inline plugins/serializers/json/README.md:105]:
```suggestion
When batch format is used and the configuration option output_json_array
```

File: plugins/serializers/json/README.md [hunk 2]
```diff
@@ -40,6 +40,9 @@ The `json` output data format converts metrics into JSON documents.
   ## can contain wildcards.
   #json_nested_fields_include = []
   #json_nested_fields_exclude = []
+
+  ## Enable json array output 
+  # output_json_array
```
Comment (DStrand1) [inline plugins/serializers/json/README.md:45]: Please set a value here so it is clear to the user that this is a boolean
```suggestion
  # output_json_array = false
```

**Review by DStrand1 — COMMENTED** [review]
Thanks for the PR! I left some comments, but could you also fix the linter issues? Thanks!

**Pull request closed without merging**